tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -200,12 +201,25 @@ def general_data_processing(self, X: DataFrame, X_val: DataFrame, X_unlabeled: D
             logger.log(20, f'Train Data Class Count: {self.label_cleaner.num_classes}')
 
         if X_val is not None and self.label in X_val.columns:
+            X_val_og = X_val
             X_val = self.cleaner.transform(X_val)
             if len(X_val) == 0:
-                logger.warning('All X_val data contained low frequency classes, ignoring X_val and generating from subset of X')
+                logger.warning('############################################################################################################\n'
+                               'WARNING: All X_val data contained low frequency classes, ignoring X_val and generating from subset of X\n'
+                               '\tYour input validation data or training data labels might be corrupted, please manually inspect them for correctness!')
+                if self.problem_type in [BINARY, MULTICLASS]:
+                    train_classes = sorted(list(y_uncleaned.unique()))
+                    val_classes = sorted(list(X_val_og[self.label].unique()))
+                    logger.warning(f'\tTraining Classes: {train_classes}')
+                    logger.warning(f'\tTuning   Classes: {val_classes}')
+                    missing_classes = [c for c in val_classes if c not in train_classes]
+                    logger.warning(f'\tClasses missing from Training Data: {missing_classes}')
```

Comment:
maybe worth adding types displayed:
```
	Training Classes: ['0', '1', '2', 'N']; type: object
	Tuning   Classes: [0, 1, 2]: type: int
```

Reply:
Another thought: it could be a problem if tuning data has different distribution of labels even if there is an overlap. I.e. training has labels [0, 1, 2] in roughly the same proportions, but tuning data only [0, 1] in 70/30 ratio - that potentially could to be a problem.

But.. we are closing EDA boundaries here.

Reply:
Nice! Added